pages/detention-locations.vue

```diff
@@ -0,0 +1,228 @@
+         descriptionShowing: false,
+         filters: {
+           "hasWiki": {
+             "label": "Has wiki",
+             "active": true,
+             "iconUrl": "/images/witch-single-blue.png"
+           },
+           "noWiki": {
+             "label": "No wiki",
+             "active": true,
+             "iconUrl": "/images/witch-single-orange.png"
+           }
+         },
+         showing: false
+       }
+     }
+   }),
+   computed:{
+   icons() {
+      const { icons } = useIcons();
+      return icons.value;
+    }
+  },
+   methods: {
+     loadWikiEntries: function () {
+       const sparqlQuery = `SELECT DISTINCT ?item ?LabelEN ?page_title
+            WHERE {
+              ?item wdt:P4478 ?witch .
+              ?article schema:about ?item ; schema:isPartOf <https://en.wikipedia.org/> ;  schema:name ?page_title .
+              ?item rdfs:label ?LabelEN filter (lang(?LabelEN) = "en") .
+            }`;
+
+       const queryDispatcher = new SPARQLQueryDispatcher(this.sparqlUrl);
+       queryDispatcher.query(sparqlQuery).then(result => {
+
+         for (let i = 0; i < result.results.bindings.length; i++) {
+           let item = result.results.bindings[i];
+
+           let wikiPage = {
+             id: item.item.value,
+             pageTitle: item.page_title.value,
+           }
+
+           this.wikiPages.push(wikiPage);
+         }
+
+       });
+     },
+     setMarkersIcons: function () {
+       let Filtering = new FilteringMethods(this.filterProperties, "sex");
+
+       for (let i = 0; i < this.originalMarkers.length; i++) {
+         let marker = this.originalMarkers[i];
+         [
+           marker.markerIcon,
+           marker.active
+         ] = Filtering.getMarkerStateIconDependant(marker);
+       }
+     },
+     hasLocalStorageExpired: function () {
+       let hours = 24; // Reset when storage is more than 24hours
+       let now = new Date().getTime();
+       let setupTime = localStorage.getItem('setupTime');
+
+       return setupTime === null || (now - setupTime > hours * 60 * 60 * 1000);
+     },
+     loadDataFromLocalStorage: function () {
+       let allFilters = JSON.parse(localStorage.getItem('allFilters'));
+       this.filterProperties.socialClass.filters = allFilters.socialClass;
+       this.filterProperties.occupation.filters = allFilters.occupation;
+     },
+     loadData: async function () {
+       this.loadWikiEntries();
+       const icons = this.icons;
+
+       try {
+         let response = await myFetch('/main.php?type=detention')
+         console.log('API Response:', response);
+         this.queryOutput = response
+       } catch (e) {
+         Swal.fire({
+           title: 'Server Error',
+           html: '<div>We are unable to connect to the server to pull in map info. Please refresh the page and try again. If this error persists, please contact <a href="mailto:ltw-apps-dev.ed.ac.uk">ltw-apps-dev.ed.ac.uk</a></div>',
```

Comment:
It would be great if the email here could be a config value we can easily change rather than hard coded in. I expect there are other places this email is referenced so would be good to change them as well if possible. Thanks Ruby

Reply:
good idea, have done this and changed for this page, should i make the changes for the other page in a new branch? if so a branch of this branch or of the branch with all  the pages added that hasnt been reviewed yet?

Reply:
Sorry I missed this comment. Yep, we would need to update this elsewhere. For any pages that have already been merged in, please create a new branch to handle them all. For the others that are awaiting merging, please update the PRs once this has been merged in. Thanks Ruby

Reply:
to create the prs for the ones already merged in should i do this after this has been merged in so that it is in the configuration?

Reply:
actually I'm sorry Ruby. It would be easier to just make the changes in this branch for the ones that are already merged in. Is that ok? Thanks!

Reply:
done

Reply:
Thank you. It looks like we also use this email address on the accessibility page. Is it possible to update it to use the config value there as well? Cheers Ruby

Reply:
done

Reply:
Thanks Ruby.